### PR TITLE
Thesis enis

### PR DESCRIPTION
**Pull request by enisze**
Merging BatchRecommender project into master
TODO: Exception handling ( If LOV (or another recommender) fails)

### CODE REVIEW

File: src/main/java/de/rwth/dbis/neologism/recommender/Query.java [hunk 1]
```diff
@@ -45,8 +50,8 @@ public Query(String query, Model context) {
     }
 
     public Query(String query, Model context, int limit) {
-        this.context = Preconditions.checkNotNull(context);
-
+        // this.context = Preconditions.checkNotNull(context);
```
Comment (miselico) [inline src/main/java/de/rwth/dbis/neologism/recommender/Query.java:53]: The context not being null is a precondition for many of the existing processors.

Comment (JohannesLipp) [reply src/main/java/de/rwth/dbis/neologism/recommender/Query.java:53]: Good point. @enisze is it possible to overload that method for your needs? That is, keep the existing method intact (with the Null check active), and add a second one with the same name for your needs, like `public Query(String query, int limit)`, which you call from your "new" lines of code.

Comment (JohannesLipp) [reply src/main/java/de/rwth/dbis/neologism/recommender/Query.java:53]: I just saw you did something like this. Still, please alternate the method calls or so in order to ensure that the Null check is in place if someone calls this `Query(query,context,limit)` constructor.

Comment (miselico) [reply src/main/java/de/rwth/dbis/neologism/recommender/Query.java:53]: This is about the contracts of the class. Just creating a new constructor does not solve this issue.

**Comment by miselico** [comment]
I don't have time for a proper code review now and the pull request is very large and modifies a lot of existing things. As also the functionality appears pretty different it likely makes more sense as a fork, rather than trying to push these changes into the existing recommender.

File: src/main/java/de/rwth/dbis/neologism/recommender/Query.java (hunk 1 above)
Comment (enisze) [reply src/main/java/de/rwth/dbis/neologism/recommender/Query.java]: this was a very old state - i returned the changes :) I am not using Query

**Comment by enisze** [comment]
> I don't have time for a proper code review now and the pull request is very large and modifies a lot of existing things. As also the functionality appears pretty different it likely makes more sense as a fork, rather than trying to push these changes into the existing recommender.

The existing recommender should still be working.

File: src/main/java/de/rwth/dbis/neologism/recommender/Query.java (hunk 1 above)
Comment (JohannesLipp) [reply src/main/java/de/rwth/dbis/neologism/recommender/Query.java]: Thank you for clarifying.

**Comment by JohannesLipp** [comment]
> I don't have time for a proper code review now and the pull request is very large and modifies a lot of existing things. As also the functionality appears pretty different it likely makes more sense as a fork, rather than trying to push these changes into the existing recommender.

@miselico We do get your point. I created a [tag](https://github.com/Semantic-Society/Recommender/tags) and a [release](https://github.com/Semantic-Society/Recommender/releases) in GitHub to make the current state persistent. I hope you agree to merge this now, while we keep all previous functionality in place.

**Pull request merged**